functions/Get-DbaMsdtc.ps1

```diff
@@ -90,77 +94,73 @@ Get DTC status for the computer srv0042 and show in a grid view
+			New-PSDrive -Name HKCR -PSProvider Registry -Root HKEY_CLASSES_ROOT | Out-Null
 			Get-ItemProperty -Path HKCR:\CID\*\Description |
 			Select-Object @{ l = 'Data'; e = { $_.'(default)' } }, @{ l = 'CID'; e = { $_.PSParentPath.split('\')[-1] } }
-			Remove-PSDrive -Name HKCR | Out-Null;
+			Remove-PSDrive -Name HKCR | Out-Null
 		}
 	}
-	PROCESS
+PROCESS
 	{
-		ForEach ($Computer in $Computername)
-		{
-			$dtcservice = $null
-			try
-			{
-				Write-Verbose "Getting DTC on $computer via WinRM"
-				$dtcservice = Get-Ciminstance -ComputerName $Computer -Query $query -ErrorAction Stop |
-				Select-Object PSComputerName, DisplayName, State, status, StartMode, StartName
-			}
-			catch
-			{
-				try
-				{
-					Write-Verbose "Failed To get DTC via WinRM. Getting DTC on $computer via DCom"
-					$SessionParams = @{ }
-					$SessionParams.ComputerName = $Computer
-					$SessionParams.SessionOption = (New-CimSessionOption -Protocol Dcom)
-					$Session = New-CimSession @SessionParams
-					$dtcservice = Get-Ciminstance -CimSession $Session -Query $query -ErrorAction Stop |
-					Select-Object PSComputerName, DisplayName, State, status, StartMode, StartName
-				}
-				catch
-				{
-					Write-Warning "Can't connect to CIM on $Computer"
-				}
-			}
-			$reg = $CIDs = $null
-			$CIDHash = @{ }
-			try
-			{
-				Write-Verbose "Getting Registry Values on $Computer"
-				$reg = Invoke-Command -ComputerName $Computer -ScriptBlock $DTCSecurity -ErrorAction Stop
-				$CIDs = Invoke-Command -ComputerName $Computer -ScriptBlock $DTCCIDs -ErrorAction Stop
-				foreach ($key in $CIDs) { $CIDHash.Add($key.Data, $key.CID) }
-			}
-			catch
-			{
-				Write-Warning "Can't connect to registry on $Computer"
-			}
-			if ($dtcservice)
-			{
-				[PSCustomObject]@{
-					ComputerName = $dtcservice.PSComputerName
-					DTCServiceName = $dtcservice.DisplayName
-					DTCServiceState = $dtcservice.State
-					DTCServiceStatus = $dtcservice.Status
-					DTCServiceStartMode = $dtcservice.StartMode
-					DTCServiceAccount = $dtcservice.StartName
-					DTCCID_MSDTC = $CIDHash['MSDTC']
-					DTCCID_MSDTCUIS = $CIDHash['MSDTCUIS']
-					DTCCID_MSDTCTIPGW = $CIDHash['MSDTCTIPGW']
-					DTCCID_MSDTCXATM = $CIDHash['MSDTCXATM']
-					networkDTCAccess = $reg.networkDTCAccess
-					networkDTCAccessAdmin = $reg.networkDTCAccessAdmin
-					networkDTCAccessClients = $reg.networkDTCAccessClients
-					networkDTCAccessInbound = $reg.networkDTCAccessInbound
-					networkDTCAccessOutBound = $reg.networkDTCAccessOutBound
-					networkDTCAccessTip = $reg.networkDTCAccessTip
-					networkDTCAccessTransactions = $reg.networkDTCAccessTransactions
-					XATransactions = $reg.XATransactions
-				}
-			}
-		}
+    ForEach ($Computer in $Computername)
+    {
+      $dtcservice = $null
+      Write-Verbose "$functionName - Getting DTC on $computer via WinRM"
+      $dtcservice = Get-Ciminstance -ComputerName $Computer -Query $query

```

Comment:
Well ... since it is supposed to go on either way, how about tying it in with the Stop-Function methodology?
Use try/catch, write a warning when it is _not_ set to silent or write an error when it is.

Thoughts anybody:
 - Manually implement this each time?
 - Integrate it into Stop-Function, even though the code continues on the next line (currently it either terminates the function or calls continue)
 - Build a dedicated internal function for this?
The first option will need no overhead, but has lots of ()small pieces of) duplication.
The second option already has the infrastructure for that, but is not inuitively named
The third option would be the cleanest, but take more implementation (I've got another idea brewing, where it could be rolled into the general verbosity/debug output management, in which case this would probably vastly outshine the second option)